### DISCUSSION TITLE
(node:26404) UnhandledPromiseRejectionWarning: Error: EPERM: operation not permitted

### DISCUSSION THREAD

**Discussion by Manikanta-20** [vercel/next.js#14719] [Help]
# Bug report
Same issue i have raised in next.js discussion ,no one provided the solution,right now ,i got same error  for different page

ERROR
(node:26404) UnhandledPromiseRejectionWarning: Error: EPERM: operation not permitted, lstat 'D:\example \.next\static\development\pages\login.js'
    at realpathSync (fs.js:1615:7)
    at deleteCache (D:\example \node_modules\next\dist\build\webpack\plugins\nextjs-require-cache-hot-reloader.js:1:177)
    at D:\example \node_modules\next\dist\build\webpack\plugins\nextjs-require-cache-hot-reloader.js:2:355
    at _next0 (eval at create (D:\example \node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:19:1)
    at eval (eval at create (D:\example \node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:32:1)
    at D:\example \node_modules\next\dist\build\webpack\plugins\unlink-removed-pages-plugin.js:2:530

when this issue comes Next.js  o/p logger will :
event - build page: /login
event - build page: /login
event - build page: /login
....

it is not  build page it will struct there? 

some time ,this issue will come for normal page also?

# What is the solution?

**Comment by paambaati** [comment]
@sai1139 Can you follow the issue template (the text you initially see when you open a new issue) and provide more code that we can run and see this issue ourselves? If we can’t reproduce this issue, it will be very difficult to help you.

**Comment by Manikanta-20** [comment]
> @sai1139 Can you follow the issue template (the text you initially see when you open a new issue) and provide more code that we can run and see this issue ourselves? If we can’t reproduce this issue, it will be very difficult to help you.

@paambaati Please check it


`import React from 'react';
import {
  TextField,
  Box,
  Typography,
  InputAdornment,
  IconButton,
  Link,
  Button,
  FormControl,
  RadioGroup,
  FormControlLabel,
  Radio,
} from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import Visibility from '@material-ui/icons/Visibility';
import VisibilityOff from '@material-ui/icons/VisibilityOff';
import Router from 'next/router';
import { loginRouter } from '../../lib/login-service';
import { sendEmail } from '../../lib/forgot-reset-password';
import useSnackBars from '../../components/toast/toast';
import { LoginFooter } from '../../components/main-footer';
import { useLoader } from '../../components/loader/load';
import ForgotUserid from './forgot-user-id';
import ForgotPasswordUi from './forgot-password';

const useStyles = makeStyles((_theme) => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  },
  label: {
    background: '#f5f5f5',
    textAlign: 'center',
    width: '100%',
    margin: '0px',
    borderBottom: '1px solid #ccc',
    padding: '8px 0px 8px 0px',
    fontWeight: 'bold',
  },
  textField: {
    width: '300px',
    marginTop: '20px',
  },
  link: {
    cursor: 'pointer',
    margin: _theme.spacing(2),
  },
  smalllabel: {
    textAlign: 'center',
    width: '100%',
    margin: '0px',
    fontSize: '14px',
    padding: '8px 0px 8px 0px',
  },
  buttons: {
    marginLeft: '35px',
  },
  modalButton: {
    marginTop: '20px',
  },
}));
export const Login = () => {
  const classes = useStyles();
  const [values, setValues] = React.useState({ showPassword: false });
  const [username, setUsername] = React.useState('');
  const [password, setPassword] = React.useState('');
  const handleClickShowPassword = () => {
    setValues({ ...values, showPassword: !values.showPassword });
  };
  const handleMouseDownPassword = (event) => {
    event.preventDefault();
  };
  const changeusername = (event) => {
    event.preventDefault();
    setUsername(event.target.value);
  };
  const changePassword = (event) => {
    event.preventDefault();
    setPassword(event.target.value);
  };
  const active = React.useMemo(() => {
    if (username.trim() === '' || password.trim() === '') return false;
    return true;
  }, [password, username]);
  const handleSubmit = async (event) => {
    event.preventDefault();
    try {
      const response = {username, password, value};
	  console.log(response);
    } catch (ex) {
      console.log(ex);
    }
  };

  const onReset = (event) => {
    event.preventDefault();
    setPassword('');
    setUsername('');
  };

  return (
    <>
      <Box
        alignItems="center"
        border={1}
        borderColor="#ccc"
        display="flex"
        borderRadius="2px"
        maxWidth="800px"
        alignSelf="center"
        margin="60px auto 0px auto"
        height="auto"
        padding="0px"
        flexDirection="column"
      >
        <Typography className={classes.label} variant="subtitle1" component="h6">
          Login 
		  </Typography>
        <form noValidate className={classes.root} onSubmit={(event) => handleSubmit(event)}>
          <TextField
            className={classes.textField}
            id="userid"
            variant="outlined"
            label={label}
            size="small"
            value={username}
            type="text"
            onChange={changeusername}
          />
          <TextField
            className={classes.textField}
            id="password"
            type={values.showPassword ? 'text' : 'password'}
            variant="outlined"
            label="Password"
            size="small"
            value={password}
            onChange={changePassword}
            InputProps={{
              endAdornment: (
                <InputAdornment position="end">
                  <IconButton
                    edge="end"
                    aria-label="toggle password visibility"
                    onClick={handleClickShowPassword}
                    onMouseDown={handleMouseDownPassword}
                  >
                    {values.showPassword ? <Visibility /> : <VisibilityOff />}
                  </IconButton>
                </InputAdornment>
              ),
            }}
          />
          <br />
          <Box flexDirection="row" marginBottom="20px">
            <Button variant="contained" color="primary" type="submit" disabled={!active}>
              LOGIN
            </Button>
            <Button variant="contained" className={classes.buttons} onClick={onReset}>
              RESET
            </Button>
          </Box>
        </form>
      </Box>
    </>
  );
};
`

**Comment by paambaati** [comment]
Okay. Now, try to answer all these questions, one by one.

1. When does this error occur? Every time you run, or when you do something specifically?
2. `EPERM` generally means a permission error. How were the dependencies installed? Do you have write permissions to the `D:\examples` directory?

**Comment by Manikanta-20** [comment]
> Okay. Now, try to answer all these questions, one by one.
> 
> 1. When does this error occur? Every time you run, or when you do something specifically?
> 2. `EPERM` generally means a permission error. How were the dependencies installed? Do you have write permissions to the `D:\examples` directory?

1) i don't know,on what based this error will come. every time i'll run same
2)Mine  OS is windows have all permission .

Some times , i have faced for some page which contains only view

**Comment by paambaati** [comment]
I’m sorry, but there isn’t enough information to help you here. Your best bet will be to either a) set up a repo where we can reproduce the issue ourselves or b) find others with the same issue and see if they’ve been able to find a solution.

**Comment by Manikanta-20** [comment]
ok,but problem is to reproduce this error.

**Comment by Manikanta-20** [comment]
@timneutkens  what is the solution?

This issue is coming from custom server?

**Comment by saqibameen** [comment]
@Manikanta-20 @paambaati I am facing the same issue in next.js development. It occurs randomly and stops compiling for further changes. Any solution?

**Reply by Manikanta-20** [reply]
@saqibameen  
Your issue is reproducible?

until now there is no solution.. to reproduce the error is challenging because it occurs randomly.  

Thanks

**Reply by saqibameen** [reply]
@Manikanta-20 Yeah, it happens randomly. There is no "exact" way to reproduce. :( 

Wastes tons of time as I have to start again every time it happens.
